src/backend/joanie/signature/backends/lex_persona.py

```diff
@@ -661,3 +662,34 @@ def update_signatories(self, reference_id: str, all_signatories: bool) -> str:
             )
 
         return response.json()["id"]
+
+    def get_signature_state(self, reference_id: str) -> dict:
+        """
+        Get the signature state progress on a signing procedure.
+        It returns a dictionary whether the student and the organization have signed.
+        """
+        timeout = settings.JOANIE_SIGNATURE_TIMEOUT
+        base_url = self.get_setting("BASE_URL")
+        token = self.get_setting("TOKEN")
+        url = f"{base_url}/api/workflows/{reference_id}/"
+        headers = {"Authorization": f"Bearer {token}"}
+
+        response = requests.get(url, headers=headers, timeout=timeout)
+
+        if not response.ok:
+            logger.error(
+                "Lex Persona: Unable to retrieve the signature procedure"
+                " the reference does not exist %s, reason: %s",
+                reference_id,
+                response.json(),
+                extra={
+                    "url": url,
+                    "response": response.json(),
+                },
+            )
+            raise exceptions.SignatureProcedureNotFound(
+                "Lex Persona: Unable to retrieve the signature procedure"
+                f" the reference does not exist {reference_id}"
+            )
+
+        return get_signature_progress_state(response.json().get("progress"))
```

Comment:
The progress is only related to Lex Persona, so I think this logic should be in its backend code.
Also, I doubt that a function is needed.

```suggestion
    # The `value` represents the progress:
    #   - 0: Neither the student nor the organization has signed.
    #   - 50: The student has signed, but the organization has not.
    #   - 100: Both the student and the organization have signed.
    value = response.json().get("progress")
    return {"student": value >= 50, "organization": value == 100}
```

Reply:
WDYT about using the `isFinished` attribute of the two steps instead of progress ?

Reply:
The `isFinished` attribute is a good way to find if depending the two different steps (step1 : student, step2: organization) has signed on their turn.
Since it returns a boolean value, which is what we want, I'm agreeing to go in this direction for that implementation and drop the function `get_signature_progress_state`.